wfe2/wfe.go

```diff
@@ -2143,6 +2145,11 @@ func (wfe *WebFrontEndImpl) GetOrder(ctx context.Context, logEvent *web.RequestE
 		return
 	}
 
+	if order.Id == 0 || order.RegistrationID == 0 || order.Expires == 0 || len(order.Names) == 0 {
```

Comment:
The fact that this check and the similar check on 2291 check for different things just goes to show how inconsistent our error checking has historically been :( Let's try to unify these now.

What fields do we actually require here? According to RFC 8555, the only fields we are `required` to surface in an Order resource are the `status` and the `finalize` url (the `identifiers` and `authorizations` are also required, but could be empty lists). So we need to ensure `status` is set, and we need `RegistrationId` and `Id` because those go into creating the finalize URL. We want to ensure `created` is non-zero so that `staleEnoughToGETOrder` does the right thing. Oh, and the RFC requires that `expires` be set in certain circumstances, so we might as well check for it all the time. And I think that's it.